servant-http-streams/test/Servant/StreamSpec.hs

```diff
@@ -0,0 +1,184 @@
+import           Test.Hspec
+
+#if MIN_VERSION_base(4,10,0)
+import           GHC.Stats
+                 (gc, gcdetails_live_bytes, getRTSStats)
+#else
+import           GHC.Stats
+                 (currentBytesUsed, getGCStats)
+#endif
+
+-- This declaration simply checks that all instances are in place.
+-- Note: this is streaming client
+_ = client comprehensiveAPI
+
+spec :: Spec
+spec = describe "Servant.Client.Streaming" $ do
+    streamSpec
+
+type StreamApi =
+         "streamGetNewline" :> StreamGet NewlineFraming JSON (SourceIO Person)
+    :<|> "streamGetNetstring" :> StreamGet NetstringFraming JSON (SourceIO Person)
+    :<|> "streamALot" :> StreamGet NoFraming OctetStream (SourceIO BS.ByteString)
+    :<|> "streamBody" :> StreamBody NoFraming OctetStream (SourceIO BS.ByteString) :> StreamGet NoFraming OctetStream (SourceIO BS.ByteString)
+
+api :: Proxy StreamApi
+api = Proxy
+
+getGetNL, getGetNS :: ClientM (SourceIO Person)
+getGetALot :: ClientM (SourceIO BS.ByteString)
+getStreamBody :: SourceT IO BS.ByteString -> ClientM (SourceIO BS.ByteString)
+getGetNL :<|> getGetNS :<|> getGetALot :<|> getStreamBody = client api
+
+alice :: Person
+alice = Person "Alice" 42
+
+bob :: Person
+bob = Person "Bob" 25
+
+server :: Application
+server = serve api
+    $    return (source [alice, bob, alice])
+    :<|> return (source [alice, bob, alice])
+    -- 2 ^ (18 + 10) = 256M
+    :<|> return (SourceT ($ lots (powerOfTwo 18)))
+    :<|> return
+  where
+    lots n
+        | n < 0     = Stop
+        | otherwise = Effect $ do
+            let size = powerOfTwo 10
+            mbs <- getHardwareEntropy size
+            bs <- maybe (getEntropy size) pure mbs
+            return (Yield bs (lots (n - 1)))
+
+powerOfTwo :: Int -> Int
+powerOfTwo = (2 ^)
+
+{-# NOINLINE manager' #-}
+manager' :: C.Manager
+manager' = unsafePerformIO $ C.newManager C.defaultManagerSettings
+
+withClient :: ClientM a -> BaseUrl -> (Either ServantError a -> IO r) -> IO r
+withClient x baseUrl' = withClientM x (mkClientEnv manager' baseUrl')
+
+testRunSourceIO :: SourceIO a
+    -> IO (Either String [a])
+testRunSourceIO = runExceptT . runSourceT
+
+streamSpec :: Spec
+streamSpec = beforeAll (CS.startWaiApp server) $ afterAll CS.endWaiApp $ do
+    it "works with Servant.API.StreamGet.Newline" $ \(_, baseUrl) -> do
+        withClient getGetNL baseUrl $ \(Right res) ->
+            testRunSourceIO res `shouldReturn` Right [alice, bob, alice]
+
+    it "works with Servant.API.StreamGet.Netstring" $ \(_, baseUrl) -> do
+        withClient getGetNS baseUrl $ \(Right res) ->
+            testRunSourceIO res `shouldReturn` Right [alice, bob, alice]
+
+    it "works with Servant.API.StreamBody" $ \(_, baseUrl) -> do
+        withClient (getStreamBody (source input)) baseUrl $ \(Right res) ->
+            testRunSourceIO res `shouldReturn` Right output
+        where
+          input = ["foo", "", "bar"]
+          output = ["foo", "bar"]
+
+{-
```

Comment:
just temporarily commented out? or is it a way to tag this as something to revisit later?

Reply:
OK, then yes I'd suggest to remove this or alternatively explain why it's commented out right at the beginning of the comment, as you prefer. (depends on how likely we are to revisit the constant-memory testing)